### ISSUE TITLE
JWT callback gets called twice when using getInitialProps/getServerSideProps

### ISSUE THREAD

**Issue by remcotak** [nextauthjs/next-auth#1674]
**Describe the bug**
When using the getSession function inside getInitialProps/getServerSideProps AND using the useSession hook in a component on the same page, the JWT callback gets called twice, even when using the provider in _app.js. When using Refresh Token Rotation flow this is a big problem since the first time the JWT callback gets called it might go into the refresh token flow and return the updated access token, but in the 2nd JWT callback the access token isn't updated with the newly received access token.

**Steps to reproduce**
Expand the next-auth tutorial code with the Refresh Token Rotation code provided here https://next-auth.js.org/tutorials/refresh-token-rotation. Use the getSession inside getServerSideProps and on the same page let a component user the useSession hook.

**Expected behavior**
I expect the useSession hook to use the data from the getSession call

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by typedashutosh** [comment]
It happens with me all time.
 But I believe it's a NextJS feature, since code is run first at compile time, and again when client requests the page.
So far, it had not cause a worry for me.

**Comment by remcotak** [comment]
I don't think it is, the first request is made through server side rendering. I await the refresh token call to return the newly received token and then the component call should be called retrieving the updated token since I'm waiting for the response to finish. I don't how else you are supposed to implement a refresh token flow.

**Comment by balazsorban44** [comment]
@remcotak Do you use Provider? https://next-auth.js.org/getting-started/client#provider

**Comment by vstrider** [comment]
I have a similar issue, but not quite the same, and it only happens at build time. Note that this only happens in dev mode, and may not be an issue in production. Let me try to describe what happens in my case:

1. Log the user in with a provider that gives back jwt and a refresh token that can be used only once
2. While the application is working, token refreshing works as intended
3. Stop the application while the user session is still alive, but leave it open in a browser
4. Wait until the token is expired and start the application back up
5. This will cause the application to execute the refresh request with a stale token that has already been consumed. `getSession()` is only invoked in `_app.getInitialProps`, and `useSession()` is not invoked at all.

I didn't want to open a separate Issue for this because I'm not sure is this a bug or just a side-effect of fast refresh or something else that just occurs during build in dev mode.
